lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -59,13 +59,15 @@ def collection_ancestor? collections
     end
 
     def ancestor? collection
-      return true if collection == self
+      return true if (( !collection.id.nil? && !self.id.nil? ) && ( collection.id == self.id ))
+      return true if (( collection.id.nil? || self.id.nil? ) && ( collection == self ))
```

Comment:
@jcoyne The purpose of the ancestor check is not to determine if the state of the two objects is equivalent, but to determine if the two objects represent the same object in Fedora.  Thus, id check is ok.  For the method in core.rb, it uses == operator which is generally considered to return false unless all aspects of the object are equal.  So in this case, you would want to do the full object equivalence test, right?

If you agree, then the changes proposed for ancestor? would still apply.